### ISSUE TITLE
Add algorithm: N-Queens

### ISSUE THREAD

**Issue by joshuaguerin** [joshuaguerin/Answer-Set-Programming-Algorithms#5]
It feels a bit obvious, but I think that is the curse of knowledge. Queens is so common because it is a great illustration. Probably going to add.

**Comment by joshuaguerin** [comment]
Almost finished. Just need to craft a diagonal constraint.

**Comment by joshuaguerin** [comment]
Observation: gotta be basic arithmetic. Subtraction and absolute value get us really close. I.e.,

```
for i in range(1, 5):
     for j in range(1, 5):
             print(abs(i-j), end = ' ')
     print()
```

gives us:
```
0 1 2 3 
1 0 1 2 
2 1 0 1 
3 2 1 0 
```

Obviously this is incorrect: clingo will gladly put them on the remaining diagonal.

**Comment by joshuaguerin** [comment]
Oh, turns out I had a mistake in my test for one of my constraints. I think i+j and i-j are actually correct here for diagonals. Turns out I was comparing both tuples for inequality rather than comparing between tuple elements (which was my intention). Fixed(?) in 8a778afde6b3b638c7a4292bde809c15f4e6cc8b?

Going to close for now knowing I'll need to test a bit. (New problem: can I make it prettier?)
